AmethystRuntime/src/AmethystRuntime.cpp

```diff
@@ -48,10 +48,19 @@ void AmethystRuntime::LoadMods() {
     if (config.promptDebugger) AttachDebugger();
 
     // Load functions from the mods
-    for each (auto mod in m_mods)
-    {
+    for each (auto mod in m_mods) {
         FARPROC addr;
 
+        addr = mod.GetFunction("GetVersion");
+        if (addr != NULL) {
+            std::string modVersion = reinterpret_cast<ModGetVersion>(addr)();
+            if(modVersion != config.version) {
+                Log::Error("[AmethystRuntime] '{0}' ({1}) does not match the current version ({2}).", mod.modName, modVersion, config.version);
+            }
+        } else {
+            Log::Warning("[AmethystRuntime] '{}' does not have 'std::string GetVersion()'. A mod should have this function for version checking to work.\n", mod.modName);
```

Comment:
I think it might be better to pass down the version in ModInitialize and then the mod can decide what to do based on that.